File: Haskell/beginner/1012.hs
-- Doing imports to use functions
import Text.Printf

-- Main Function
main = do
    -- Requesting values and assign into variables
    {-a <- readLn :: IO Float
    b <- readLn :: IO Float
    c <- readLn :: IO Float
    d <- readLn :: IO Float-}

    -- lol, I always forget to use getLine in Haskell...
    usr_expression <- getLine

    -- Now, we can use this to extract values! Aha! <3
    let [a, b, c] = map read (words usr_expression) :: [Double] -- Ok, again, I forget DOUBLE type.

    -- Doing the calculations
    let triangle = (a * c) / 2
    let circle = 3.14159 * (c ^ 2)
    let trapezium = ((a + b) * c) / 2
    let square = b * b
    let rectangle = a * b
    
    -- Showing results
    printf "TRIANGULO: %.3f\n" triangle
    printf "CIRCULO: %.3f\n" circle
    printf "TRAPEZIO: %.3f\n" trapezium
    printf "QUADRADO: %.3f\n" square
    printf "RETANGULO: %.3f\n" rectangle

-- By Fernando Serra
-- https://github.com/fernandosserra

File: C/beginner/1012.c
/*Including libraries*/
#include <stdio.h>
#include <math.h>

/*Main Function*/
int main (){
    /*Variable declaration and initialization*/
    double A, B, C;
    double TRIANGULO, CIRCULO, TRAPEZIO, QUADRADO, RETANGULO;
    
    /*Requesting values to user*/
    scanf("%lf" "%lf" "%lf", &A, &B, &C);
    
    /*Doing the calculations*/
    if (A > 0 && B > 0 && C > 0){
        TRIANGULO = (A * C) / 2;
        CIRCULO = 3.14159 * (pow(C, 2));
        TRAPEZIO = ((A + B) * C) / 2;
        QUADRADO = B * B;
        RETANGULO = A * B;
        
    }

    /*Showing results*/
    printf("TRIANGULO: %.3lf\n", TRIANGULO);
    printf("CIRCULO: %.3lf\n", CIRCULO);
    printf("TRAPEZIO: %.3lf\n", TRAPEZIO);
    printf("QUADRADO: %.3lf\n", QUADRADO);
    printf("RETANGULO: %.3lf\n", RETANGULO);    

    /*Ending code*/
    return 0;
}

/*By Fernando Serra*/
/*https://github.com/fernandosserra*/